src/cli/cli.cpp

```diff
@@ -1199,25 +1199,31 @@ template <> otError Interpreter::Process<Cmd("coex")>(Arg aArgs[])
             {&otRadioCoexMetrics::mNumRxGrantNone, "Grant None"},
         };
 
+        static otRadioCoexMetrics sLastTimeMetrics;
+
         otRadioCoexMetrics metrics;
 
         SuccessOrExit(error = otPlatRadioGetCoexMetrics(GetInstancePtr(), &metrics));
 
         OutputLine("Stopped: %s", metrics.mStopped ? "true" : "false");
-        OutputLine("Grant Glitch: %u", metrics.mNumGrantGlitch);
+        OutputLine("Grant Glitch: %u (%u)", metrics.mNumGrantGlitch, sLastTimeMetrics.mNumGrantGlitch);
         OutputLine("Transmit metrics");
 
         for (const RadioCoexMetricName &metric : kTxMetricNames)
         {
-            OutputLine(kIndentSize, "%s: %u", metric.mName, metrics.*metric.mValuePtr);
+            OutputLine(kIndentSize, "%s: %u (%u)", metric.mName, metrics.*metric.mValuePtr,
+                       sLastTimeMetrics.*metric.mValuePtr);
         }
 
         OutputLine("Receive metrics");
 
         for (const RadioCoexMetricName &metric : kRxMetricNames)
         {
-            OutputLine(kIndentSize, "%s: %u", metric.mName, metrics.*metric.mValuePtr);
+            OutputLine(kIndentSize, "%s: %u (%u)", metric.mName, metrics.*metric.mValuePtr,
+                       sLastTimeMetrics.*metric.mValuePtr);
         }
+
+        memcpy(&sLastTimeMetrics, &metrics, sizeof(metrics));
```

Comment:
@Irving-cl  I am not sure about this model of including last value in the CLI output.
- It makes coex counters different from all other OT counters
- Counters are expected to give values at the time user request them (including last value (from last time it was requested) can be confusing for new users).
- User should already know the last value from last time she/he called the `cli coex metrics`.

Reply:
Thanks! What about providing a difference value? (between current value and last value) Actually this is the main purpose.

Currently we haven't specified if the coex counters are 'accumulated' or are reset by each query. But our preference is to make it 'accumulated'. Then when we tried to know the counters for the recent period, it's inconvenient. (We have to record them each time and subtract them manually).

Do you have any suggestions? @abtink

Reply:
All other OT counters are accumulative, so make sense for coex to follow the same model.

For some counetrs we provide seperate API to clear/reset the counters -> `otLinkResetCounters`.

Resetting the counter on "read" can be somewhat limiting (we indirectly limit the user/caller to be one entity). If two entities (e.g. telemetry reading it periodically and user reading it using cli) read the counter, it can become confusing for the user.

Reply:
Ok, then maybe let's specify in the API doc that Coex Counters are accumulative and add an API to clear the counters.